x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowNodeTask.java

```diff
@@ -59,29 +66,56 @@
     private final TimeValue retryTimeout;
     private final TimeValue idleShardChangesRequestDelay;
     private final BiConsumer<TimeValue, Runnable> scheduler;
+    private final LongSupplier relativeTimeProvider;
 
-    private volatile long lastRequestedSeqno;
-    private volatile long leaderGlobalCheckpoint;
+    private final ReleasableLock readLock;
+    private final ReleasableLock writeLock;
+
+    {
```

Comment:
I don't like the locking semantics - read/ write for stats updates and fully synced for class execution. I think we should stick with one. I understand you did it to make sure stats reading is consistent. My vote goes to the simplicity of synchronized access and making the status creation method synchronized as well. It's super light.